src/plugins/plugin.autoplay.js

```diff
@@ -1,128 +1,125 @@
-/*global BookReader */
+// @ts-check
+import { EVENTS } from "../BookReader/events";
+import { parseAnimationSpeed } from "../BookReader/utils";
+import { BookReaderPlugin } from "../BookReaderPlugin";
 
 /**
  * Plugin which adds an autoplay feature. Useful for kiosk situations.
  */
-jQuery.extend(BookReader.defaultOptions, {
-  enableAutoPlayPlugin: true,
-});
-
-/**
- * @override BookReader.setup
- */
-BookReader.prototype.setup = (function(super_) {
-  return function (options) {
-    super_.call(this, options);
-
-    this.autoTimer = null;
-    this.flipDelay = 5000;
-  };
-})(BookReader.prototype.setup);
+export class AutoplayPlugin extends BookReaderPlugin {
+  options = {
+    enabled: true,
+    /**
+     * @type {number | 'fast' | 'slow'}
+     * How quickly the flip animation should run.
+     * Defaults to global BookReader flipSpeed option if unspecified
+     **/
+    flipSpeed: null,
```

Comment:
Make autoflip flipSpeed slower than default bookreader's to match IA. Also provides a nice experience if it's a bit slower!
```suggestion
    /**
     * @type {number | 'fast' | 'slow'}
     * How quickly the flip animation should run.
     **/
    flipSpeed: 2000,
```